### ISSUE TITLE
Epic: Add Angular Route Guards to manage navigation in app

### ISSUE THREAD

**Issue by Michiel-s** [AmpersandTarski/prototype#142]
In our frontend application we have two use cases that request for Route Gard implementation:
- Ask user to save data when moving away from component when there are still uncommitted changes
- Allow access to components based on `Role` based access constraint that come from Ampersand compiler

There are five different types of Route Guards:
1. CanActivate
1. CanActivateChild
1. CanLoad
1. CanDeactivate (can be used to prevent navigating away)
1. Resolve

 Note that a Route Guard can be applied to a route itself using the `canActivate` property and/or be applied to all child routes using the `canActiveChild` property.
```ts
{
   path: 'foo',
   canActivate: [AuthGuard],
   canActivateChild: [AuthGuard],
   component: FooComponent,
   children: [
    { path: ':id', component: BarComponent},
    { path: ':id/name', component: BazComponent}
]}
```

This feature requires the implementation of keeping track of `sessionRoles` in the frontend. I'll create a separate issue for that.

EDIT:

It is not necessary for the sessionRoles to be tracked in the front-end, since this has already been handled by the back-end.

To complete this Epic, there are two issues that have to be resolved. One involves routing based on the roles of the user AmpersandTarski/prototype-frontend#177, and the other involves the user going to another route when the changes have yet to be commited AmpersandTarski/prototype#136 .
- [ ] AmpersandTarski/prototype-frontend#177 
- [ ] AmpersandTarski/prototype#136

**Comment by stefjoosten** [comment]
Is this issue still relevant? @Michiel-s can we close this?

**Comment by Michiel-s** [comment]
Yes. This issue is still relevant.  The developed RolesGuard is not activated for generated module and also the pending changes detection is open issue.
